java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -174,14 +174,16 @@ public void setInitialCapacity(int valueCount) {
    * @param valueCount desired number of elements in the vector
    * @param density average number of bytes per variable width element
    */
+  @Override
   public void setInitialCapacity(int valueCount, double density) {
-    final long size = (long) (valueCount * density);
-    if (size < 1) {
-      throw new IllegalArgumentException("With the provided density and value count, potential capacity of the data buffer is 0");
-    }
+    long size = (long) (valueCount * density);
     if (size > MAX_ALLOCATION_SIZE) {
       throw new OversizedAllocationException("Requested amount of memory is more than max allowed");
     }
+
+    if(size == 0) {
```

Comment:
I'm not sure if this is all that useful, but this means we can't have an initial capacity of 0?  Is that something that we should still consider?

Reply:
Yes we cannot have an initial capacity of 0 because then our safe* functions runs into an infinite loop where they try to realloc and have the target buffer size as next power of 2 -- BaseAllocator.nextPowerOfTwo returns 0 for 0 and thus safe functions keep calling realloc.

This happens if the initial capacity was 0.